src/model_DAforcing.py

```diff
@@ -354,13 +354,16 @@ def run(self, values: dict):
         """
         output_parameters = self._output_parameters
         da_parameters = self._data_assimilation_parameters
-
-        if output_parameters['lite_restart'] is not None:
-            _write_lite_restart(
-                values,
-                self._t0,
-                output_parameters['lite_restart']
-            )   
+        q0 = values.get('q0')
+        waterbody_df = values.get('waterbody_df')
+        lastobs_df = values.get('lastobs_df')
+        if len(q0) != 0 and len(waterbody_df) != 0 and len(lastobs_df) != 0:
```

Comment:
I don't follow what this `if` statement is for? `q0` here should never be empty, it's the main output from t-route. `waterbody_df` could be empty if waterbodies are not being simulated, but `_write_lite_restart` already checks for this before trying to write the waterbody_restart_file. I'm not sure why `lastobs_df` is being checked here.